client/state/selectors/is-gmb-nudge-visible.js

```diff
@@ -7,18 +7,56 @@ import createSelector from 'lib/create-selector';
 import { getSiteOption, getSitePlanSlug } from 'state/sites/selectors';
 import { PLAN_BUSINESS } from 'lib/plans/constants';
 
+const WEEK_IN_SECONDS = 60 * 60 * 24 * 7;
+
 /**
- * Returns true if the Google My Business (GMB) nudge should be visible in stats
+ * Returns true if site has promote goal set
  *
  * @param  {Object}  state  Global state tree
  * @param  {String}  siteId The Site ID
- * @return {Boolean} True if we should show the nudge
+ * @return {Boolean} True if site has "promote" goal
  */
-export default createSelector(
+const siteHasPromoteGoal = createSelector(
 	( state, siteId ) => {
 		const siteGoals = ( getSiteOption( state, siteId, 'site_goals' ) || '' ).split( ',' );
-		const sitePlan = getSitePlanSlug( state, siteId );
-		return siteGoals.indexOf( 'promote' ) !== -1 && sitePlan === PLAN_BUSINESS;
+		return siteGoals.indexOf( 'promote' ) !== -1;
 	},
-	( state, siteId ) => [ getSitePlanSlug( state ), getSiteOption( state, siteId, 'site_goals' ) ]
+	( state, siteId ) => [ getSiteOption( state, siteId, 'site_goals' ) ]
+);
+
+/**
+ * Returns true if site has business plan
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {String}  siteId The Site ID
+ * @return {Boolean} True if site has business plan
+ */
+const siteHasBusinessPlan = createSelector(
+	( state, siteId ) => getSitePlanSlug( state, siteId ) === PLAN_BUSINESS,
+	( state, siteId ) => [ getSitePlanSlug( state, siteId ) ]
 );
+
+/**
+ * Returns true if the Google My Business (GMB) nudge should be visible in stats
+ *
+ * It should be visible if:
+ * - site is older than 1 week,
+ * - site has a business plan
+ * - site has a promote goal
```

Comment:
There is another condition to take into account: if the user dismisses the nudge, we should display it again **only one more** time **two weeks** after that event.

Reply:
That's handled via `DismissibleCard`

Reply:
> We have no way of knowing how many times we showed the nudge to a user. And if we plan to add that as a feature, it will probably need to be persisted on the backend. In any case it should not be part of this PR.

Apologies if I'm missing something, but as far I can see we already store when the nudge was dismissed [in the backend](https://github.com/Automattic/wp-calypso/blob/5f15f3365213253724ca0d8798c6cb8160f00bc9/client/state/preferences/actions.js#L95-L112). This is why we should consider @cbauerman's proposal about storing an array of timestamps instead of a single one.

> Imo we can do without since we'll already be working on M2 in 2 weeks so we won't be showing the nudge forever.

M2 needs to be refined, but we may very well keep the same behavior for the nudge.

Reply:
I'm not sure I understand the use case for multiple timestamps, if it's for statistics, we have tracks for that... Why do we need it?